compleet-client/src/ui/menu.rs

```diff
@@ -174,8 +175,26 @@ pub fn find_position(
 /// Returns the number of screen rows above and below the current cursor
 /// position.
 fn rows_above_below_cursor(lua: &Lua) -> LuaResult<(u16, u16)> {
-    let total_rows = api::get_option::<u16>(lua, "lines")?;
-    let rows_above = r#fn::screenrow(lua)? - 1;
+    let lines = api::get_option::<u16>(lua, "lines")?;
+    let cmdheight = api::get_option::<u16>(lua, "cmdheight")?;
+    let laststatus = api::get_option::<u16>(lua, "laststatus")?;
+    let tabline = api::get_option::<String>(lua, "tabline")?;
 
-    Ok((rows_above, total_rows - rows_above - 1))
+    let (screenrow, _) = crate::utils::get_screen_cursor(lua)?;
+
+    let layout = r#fn::winlayout(lua)?;
+    let is_split = layout.get::<_, String>(1)? != "leaf";
+
+    let statuslineoffset = match laststatus {
+        0 => 0,
+        1 if !is_split => 0,
+        _ => 1,
+    };
+
```

Comment:
You only need to check for splits when laststatus is 1.  Refactor this to
```rust
let statusline_offset = match laststatus {
    0 => 0,
    1 => {
        let is_split = r#fn::winlayout(lua)?.get::<_, String>(1)? != "leaf";
        if is_split { 1 } else { 0 }
    },
    _ => 1,
}

Reply:
Yeah, this is much cleaner. My bad